### DISCUSSION TITLE
Migrate Stripe account to another

### DISCUSSION THREAD

**Discussion by inventionsbyhamid** [pay-rails/pay#1063] [Q&A] [answered]
Hi. I have Stripe integrated via pay gem. I have been using a Stripe account and now we are migrating to a different Stripe account for some xyz reason. 

How do i do it smoothly? I want users to cancel subscriptions in their previous account and subscribe again (yeah i know bad customer experience and might result in a lot of churn)

I want to know if anyone has done this in the past. Need to know what options do I have. Any help or guidance would be great. I saw some Stripe FAQs related this but they don't seem to be sufficient.

**Comment by deanpcmad** [comment]
I'm not sure if Pay can work with multiple Stripe accounts in the same app 🤔

**Comment by excid3** [comment, marked as answer]
You might be able to by overriding the `stripe_options` method in Pay::Stripe::Subscription.

https://github.com/pay-rails/pay/blob/main/app/models/pay/stripe/subscription.rb#L377-L379

Stripe's Ruby gem now allows you to pass the API key in the second set of arguments, which you could use for the old account's API key. That allows you to override the default API key that's set globally.

See the Stripe docs: https://docs.stripe.com/api/authentication

**Reply by deanpcmad** [reply]
Oh that's handy, didn't know they added that

**Reply by excid3** [reply]
They used to have a Client you could create or something, but looks like they settled on this since they're using it for other functionality too.

**Reply by inventionsbyhamid** [reply]
One way I was thinking to copy existing pay tables to some with prefix "old" and use fresh pay tables for the new account.  Doing this if a pay customer exists in old tables I could show them billing portal for the old account. If not i assume they are new users and act accordingly. Obviously this is probably very ugly way to do this. The number of users is not too high.

Overriding stripe_options is looking complex to me, not sure what other changes i would be required to make. Will go through this as well.

**Reply by excid3** [reply]
You could add a column to the model to denote if it was new or old and then have it choose that API key accordingly?

**Reply by inventionsbyhamid** [reply]
Yes, I could do it like that as well. Would require changes to code everywhere I suppose/or override Pay methods, for example I was getting link to the billing portal using 
`current_user.payment_processor.billing_portal`
